### PR TITLE
PT-2470: Hide empty list of genes from export pdf and patient form when in view mode.

### PR DESCRIPTION
**Pull request by veronikaslc**
Fixed.

### CODE REVIEW

File: components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetMacros.xml [hunk 1]
```diff
@@ -257,6 +257,5 @@
     )))
   #end
 #end
-{{/velocity}}
-</content>
+{{/velocity}}</content>
```
Comment (marta-) [inline components/patient-data/ui/src/main/resources/PhenoTips/PatientSheetMacros.xml]: I would have put white space changes in a different commit, since they're not  directly related to the issue.

File: components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__gene-variants.xml [hunk 2]
```diff
@@ -112,14 +112,19 @@
 
 {{include reference="PhenoTips.TabelarDataMacros" /}}
 
-{{include reference="PhenoTips.GeneVariantMacros" /}}
```
Comment (marta-) [inline components/patient-data/ui/src/main/resources/PhenoTips/UIX_Field__gene-variants.xml]: Same for file reorganization

**Comment by marta-** [comment]
All looks good otherwise.

File: components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js [hunk 3]
```diff
@@ -3,35 +3,37 @@ var ExtraGeneVariantData = (function (ExtraGeneVariantData) {
   tools.Editor = Class.create({
 
     initialize : function () {
-      var geneTableListId = $$('.gene-table.extradata-list')[0].id;
-      this.geneClassName = geneTableListId.substring(geneTableListId.lastIndexOf('-') + 1);
-      this.geneVariantClassName = 'PhenoTips.GeneVariantClass';
```
Comment (sdumitriu) [inline components/widgets/resources/src/main/resources/resources/uicomponents/widgets/extraGeneVariantData.js]: This line was removed, and now `geneVariantClassName` is no longer defined anywhere, although it is still used in several places.
